### PR TITLE
Add particle class creation functions

### PR DESCRIPTION
**Pull request by streeve**
Includes option for passing input class directly (example in crack branching)

Unrelated change: unneeded empty prenotch in fragmentation problem

### CODE REVIEW

**Comment by streeve** [comment]
Reminder for review @pabloseleson

File: src/CabanaPD_Particles.hpp [hunk 1]
```diff
@@ -780,6 +781,54 @@ class Particles<MemorySpace, PMB, TemperatureDependent, Dimension>
 #endif
 };
 
+template <typename MemorySpace, typename ModelType, typename ExecSpace>
+auto createParticles( ExecSpace exec_space, CabanaPD::Inputs inputs )
+{
+    std::array<double, 3> low_corner = inputs["low_corner"];
```
Comment (pabloseleson) [inline src/CabanaPD_Particles.hpp:787]: Lines 787-794 and not in the createParticles function in Lines 803 and 817. How does this work?

Comment (streeve) [reply src/CabanaPD_Particles.hpp:787]: Those functions accept all these values as separate inputs instead of `CabanaPD::Inputs inputs`

Comment (pabloseleson) [reply src/CabanaPD_Particles.hpp:787]: What about the computation of m, which is not an input?

    int m = std::floor( delta /
                        ( ( high_corner[0] - low_corner[0] ) / num_cells[0] ) );
    int halo_width = m + 1; // Just to be safe.

Comment (pabloseleson) [reply src/CabanaPD_Particles.hpp:787]: Never mind. I found we compute that in the example itself.

**Review by pabloseleson — APPROVED** [review]

File: src/CabanaPD_Particles.hpp [hunk 2]
```diff
@@ -780,6 +781,54 @@ class Particles<MemorySpace, PMB, TemperatureDependent, Dimension>
...
+
+template <typename MemorySpace, typename ModelType, typename ExecSpace,
+          std::size_t Dim>
+auto createParticles( const ExecSpace& exec_space,
+                      std::array<double, Dim> low_corner,
+                      std::array<double, Dim> high_corner,
+                      const std::array<int, Dim> num_cells,
+                      const int max_halo_width )
+{
+    return std::make_shared<
```
[the remark this answers is not in the document]
Comment (streeve) [reply src/CabanaPD_Particles.hpp:809]: We cannot name a function the same thing as an existing class

**Pull request merged**